### ISSUE TITLE
state_verbose False summary is wrong

### ISSUE THREAD

**Issue by hvnsweeting** [saltstack/salt#24009]
when set to True

````
Summary
-------------
Succeeded: 34 (changed=3)
Failed:     0
-------------
Total states run:     34
````

When set to False

````
Summary
------------
Succeeded: 3 (changed=3)
Failed:    0
------------
Total states run:     3
````

I think it should be consistent with the output, it should be always as when state_verbose = True

**Comment by tbaker57** [comment]
I agree. And I'd love to submit a pull-request to fix this but the highstate outputter code is scary and I wouldn't want to break anything. 

I think the main issue is that if state_verbose=False, the return data is cleaned very early on in 
https://github.com/saltstack/salt/blob/develop/salt/output/highstate.py#L119
```
if not __opts__.get('state_verbose', False):
      data = _strip_clean(data)
```
an operation which removes all the state results which had no changes and were successful

If you instead make this check when the output is displayed e.g. (patch for 2014.7.0)
```
@@ -93,10 +93,6 @@
             hstrs.append((u'{0}----------\n    {1}{2[ENDC]}'
                           .format(hcolor, err, colors)))
     if isinstance(data, dict):
-        # Strip out the result: True, without changes returns if
-        # state_verbose is False
-        if not __opts__.get('state_verbose', False):
-            data = _strip_clean(data)
         # Verify that the needed data is present
         for tname, info in data.items():
             if '__run_num__' not in info:
@@ -123,6 +119,10 @@
                ret['result'] and not schanged:
                 continue
 
+            if not __opts__.get('state_verbose', False) and \
+               ret['result'] and not schanged:
+                continue
+
             if schanged:
                 tcolor = colors['CYAN']
             if ret['result'] is False:

````
it seems to give what you are looking for (at least in my limited tests)

**Comment by anlutro** [comment]
Funnily enough, I found this issue as I was finishing up the pull request with almost the exact patch mentioned here.

**Comment by jfindlay** [comment]
@anlutro, thanks for fixing this.  @hvnsweeting and @tbaker57, will you confirm when you can if this issue is fixed?  It will be in 2015.5.2, which will come out in two weeks.  Thanks.

**Comment by tbaker57** [comment]
@jfindlay - yes, all looks good as at 2015.5.1-110-g51c3cec. I must say, this is the first time I've used 2015.5 in earnest and it's outrageously fast for no-change runs compared to 2014.7.* - kudos to all!

cheers,
Tim
